File: perl/j11.arguments.pl
#!/usr/bin/perl
# acquire num of args
$arg_count = $#ARGV + 1; # get num of arguments
$last      = $#ARGV;     # get index of last element
$first     = 0;          # set index of first element 

print "The arguments passed are:\n";
# iterative loop to enumerate args
for ( my $count = $first; $count <= $last ; $count++ ) {
  printf "  item %d: %s\n", $count+1, $ARGV[$count];
}

File: python/j11.arguments.py
#!/usr/bin/python
import sys
# illustrative variables 
arg_count = len(sys.argv) - 1 # get num of real arguments

print "The arugments passed are:"
# use collection loop with range to enumerate arguments
#   Note: range ending must be one greater than desired range
for count in range(1,arg_count+1): # range to gen sequence of numbers
   arg = sys.argv[count]          # get arg using index
   # print count and argument using count index
   print " item %d: %s" % (count,arg) 
